pkg/libinit/root_linux.go

```diff
@@ -240,11 +241,15 @@ var (
 	}
 )
 
-// InstallAllModules installs kernel modules (.ko files) from /lib/modules.
+// InstallAllModules installs kernel modules (.ko files) from /lib/modules
+// if the kernel has loadable module support.
 // Useful for modules that need to be loaded for boot (ie a network
 // driver needed for netboot). It skips over blacklisted modules in
 // excludedMods.
 func InstallAllModules() {
+	if _, err := os.Stat("/proc/modules"); errors.Is(err, os.ErrNotExist) {
+		return
```

Comment:
Do you think log.Printf sth before return, would be helpful ?

Reply:
no. This is a package. Packages should return errors, not print.

Reply:
It would be good to log module name before each module load. In case kernel crashes, it could help figure out which module was at fault.